### DISCUSSION TITLE
How should I add another dependency when one dependency version is greater than a certain value?

### DISCUSSION THREAD

**Discussion by uf-fipper** [python-poetry/poetry#8879] [Q&A]
If `pydantic` version >= 2.0, then require `pydantic-settings`, otherwise not require. How can I do this?

**Comment by clintonroy** [comment]
I don't believe python wheels have the concept of optional dependencies like this, only extras.

**Comment by uf-fipper** [comment]
Okay, thank you all. At present, it seems impossible to achieve this...
